frontend/src/state/GraphState.jsx

```diff
@@ -573,8 +579,8 @@ function setOp(op, setA, setB) {
     difference: differenceWith,
     intersection: intersectionWith
   })[op];
-  if (setA instanceof Set || setA instanceof Iterator) setA = [...setA];
-  if (setB instanceof Set || setB instanceof Iterator) setB = [...setB];
+  setA = decomposeToArray(setA);
```

Comment:
## Bug fix: `Iterator` undefined
Apparently, `Iterator` is [only native to Firefox](https://stackoverflow.com/a/12079454/5258518). [ChatGPT helped me refactor](https://chatgpt.com/share/d9555ff7-eece-44ee-8930-6eaa28493f0a).

I also refactored by abstracting to a function:

```js
function decomposeToArray(setOrIterator) {
  if (setOrIterator instanceof Set || (setOrIterator && typeof setOrIterator.next === 'function')) setOrIterator = [...setOrIterator];
  return setOrIterator;
}
```